app/src/org/commcare/android/resource/installers/XFormAndroidInstaller.java

```diff
@@ -76,8 +77,10 @@ protected int customInstall(Resource r, Reference local, boolean upgrade) throws
 
         FormDef formDef;
         try {
-            formDef = new XFormParser(new InputStreamReader(local.getStream(), "UTF-8")).parse();
-        } catch (XFormParseException xfpe) {
+            XFormParser parser = new XFormParser(new InputStreamReader(local.getStream(), "UTF-8"));

```

Comment:
Fair. Although there are also a lot of places where an XFormParser is instantiated but we're not registering the extensions with it. Doesn't really change what you said but made me realize that I'm not sure what all of those other places are doing and why they don't need the extensions?